react/testing.md

```diff
@@ -1,3 +1,291 @@
 # Eventbrite React Testing Best Practices
 
-Coming soon...
+Guidelines and best practices used by Eventbrite to provide consistency and prevent errors in testing React components. This does not cover testing utility/helper functions (including Redux reducers) that are used in conjunction with React as those follow general testing guidelines.
+
+## Table of Contents
+
+0. [Testing environment](#testing-environment)
+0. [Testing philosophy](#testing-philosophy)
+0. [Finding nodes](#finding-nodes)
+0. [Finding components](#finding-components)
+0. [Testing existence](#testing-existence)
+0. [Assertion helpers](#assertion-helpers)
+0. [Types of renderers](#types-of-renderers)
+0. [Testing render](#testing-render)
+0. [Testing events](#testing-events)
+0. [Testing state](#testing-state)
+0. [Testing re-renders](#testing-re-renders)
+
+## Testing environment
+
+Eventbrite uses [`chai`](http://chaijs.com) (`expect` [BDD style](http://chaijs.com/api/bdd/)), [`enzyme`](http://airbnb.io/enzyme/) and [`sinon`](http://sinonjs.org/) for unit testing React components. We also leverage [`chai-enzyme`](https://github.com/producthunt/chai-enzyme) and [`sinon-chai`](https://github.com/domenic/sinon-chai) assertion helpers. Enzyme wraps [`ReactTestUtils`](https://facebook.github.io/react/docs/test-utils.html), which contains a bunch of primitives for testing components. Don't use `ReactTestUtils` directly; use Enzyme!
+
+**[⬆ back to top](#table-of-contents)**
+
+## Testing philosophy
+
+**[⬆ back to top](#table-of-contents)**
+
+## Finding nodes
+
+Search for nodes within a component by adding `data-spec` attributes to them. In the past, Eventbrite used special `js-*` CSS classes for references to nodes in JavaScript code. These `js-*` classes were also used when testing as well. Now with React testing, instead of using special CSS classes, [refs](https://github.com/eventbrite/javascript/tree/master/react#refs), or attempting to traverse the DOM with Enzyme's [`find`](http://airbnb.io/enzyme/docs/api/ReactWrapper/find.html) helper, we use `data-spec` attributes.
```

Comment:
These `js-*` classes were also used when testing as well.

remove the `as well`